__tests__/serializers/slateToReact/index.spec.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { ReactElement, ReactNode, JSXElementConstructor } from 'react'
+import { Element } from "domhandler"
+import { getName } from 'domutils'
+import {render, screen} from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { SlateToReact } from '../../../src/serializers/slateToReact'
+
+describe("React conversion", () => {
+  test("convert domhandler element to React element", async () => {
+    const slate = [
+      {
+        children: [
+          {
+            text: "Paragraph",
+          },
+        ],
+        type: 'p',
+      },
+    ]
+
+    render(<SlateToReact node={slate} />)
+
+    expect(screen.getByText('Paragraph').tagName).toBe('P')
+  })
+
+  it('can handle inline code tags', () => {
+    const html =
+      '<p>This is editable <strong>rich</strong> text, <i>much</i> better than a <pre><code>&lt;textarea&gt;</code></pre>!</p>'
+    const slate = [
+      {
+        type: 'p',
+        children: [
+          {
+            text: 'This is editable ',
+          },
+          {
+            text: 'rich',
+            bold: true,
+          },
+          {
+            text: ' text, ',
+          },
+          {
+            text: 'much',
+            italic: true,
+          },
+          {
+            text: ' better than a ',
+          },
+          {
+            text: '<textarea>',
+            code: true,
+          },
+          {
+            text: '!',
+          },
+        ],
+      },
+    ]
+    render(<SlateToReact node={slate} />)
+
+    expect(screen.getByText('much').tagName).toBe('I')
+  })
+})
```

Comment:
Test passes with a warning:

```
console.error
    Warning: validateDOMNesting(...): <pre> cannot appear as a descendant of <p>.
        at pre
        at p
        at SlateToReact (/<project-dir>/src/serializers/slateToReact/index.tsx:20:3)
```

Hadn't thought of that! However, this is acceptable given that one of the reasons that [this project uses `htmlparser2`](/docs/engineering.md#htmlparser2) is to process/render HTML in a forgiving manner.